packages/neos-ui-error/src/container/FlashMessages/FlashMessage.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the Neos.Neos.Ui package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+import React from 'react';
+import mergeClassNames from 'classnames';
+
+import {IconButton, Icon} from '@neos-project/react-ui-components';
+
+import {Severity} from '../../types';
+
+import style from './style.module.css';
+
+export const FlashMessage: React.FC<{
+    id: string;
+    message: string;
+    severity: Severity;
+    timeout?: number;
+
+    onClose: (id: string) => void;
+}> = (props) => {
+    const {message, severity} = props;
+    const isSuccess = severity === 'success';
+    const isError = severity === 'error';
+    const isInfo = severity === 'info';
+    const isClosing = React.useRef(false);
+    const handleClose = React.useCallback(() => {
+        const {onClose, id} = props;
+
+        if (!isClosing.current) {
+            isClosing.current = true;
+            setTimeout(() => onClose(id), 100);
```

Comment:
i do not quite grasp this timeout here and also was wondering if its ideal that we delegate this timeout logic to this nested component instead of handling it in a higher level ... but please dont see this as critique but rather me trying to learn the tricks ;)

Reply:
I've abstained from refactoring this stuff. This is only copied over from the original implementation and I agree that it's a bit smelly, but I didn't want to do more than strictly necessary to achieve the architectural goals.

Reply:
Perfect i oversaw that. In that case im even more grateful that its just moved as this should simplify the review :)